service/src/main/java/org/mskcc/smile/service/impl/SampleServiceImpl.java

```diff
@@ -65,15 +65,23 @@ public SmileSample saveSmileSample(SmileSample
             if (sampleHasMetadataUpdates(existingMetadata, sampleMetadata,
                     sample.getSampleCategory().equals("research"))) {
                 LOG.info("Found updates to persist for sample: " + existingSample.getPrimarySampleAlias());
-                existingSample.updateSampleMetadata(sample.getLatestSampleMetadata());
+                SampleMetadata newSampleMetadata = sample.getLatestSampleMetadata();
+                existingSample.updateSampleMetadata(newSampleMetadata);
```

Comment:
I'm curious about this - was it not working properly?

Reply:
It is but later in the code i was referring to `newSampleMetadata` multiple times so i defined it separately

Reply:
Actually couldn't you just reference `sampleMetadata` since that should be the new/incoming version?

Reply:
Yes you're right!